### PR TITLE
Add slurm-gcp v5 login node module

### PR DESCRIPTION
**Pull request by heyealex**
Adds the last module to support comparable functionality in slurm-gcp v5 with slurm-gcp v4 in the hpc toolkit. This does not update any examples, however a sample blueprint can be found in test_configs for testing. 

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 1]
```diff
@@ -0,0 +1,284 @@
+/**
```
Comment (nick-stroud) [inline community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:1]: One pattern I see is that it seems that perhaps the underlying slurm module has taken over more intelligent defaults  that we can rely on, but as a side effect the defaults of our module have become more opaque. I am not quite sure what to do about this. Any thoughts?

Maybe more prominent linking to their documentation for information on defaults?

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+  description = "Cluster name, used for resource naming and slurm accounting."
+
+  validation {
+    condition     = can(regex("(^[a-z][a-z0-9]*$)", var.slurm_cluster_name))
+    error_message = "Variable 'slurm_cluster_name' must be composed of only alphanumeric values and begin with a leter. regex: '(^[a-z][a-z0-9]*$)'."
+  }
+}
+
+variable "controller_instance_id" {
+  description = "The controller instance template"
```
[the remark this answers is not in the document]
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:45]: This is best reference I could find: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#instance_id

More clarity has been added to this and the controller output descriptions in a future commit.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+}
+
+variable "static_ips" {
+  type        = list(string)
+  description = "List of static IPs for VM instances."
+  default     = []
+}
+
+variable "access_config" {
+  description = "Access configurations, i.e. IPs via which the VM instance can be accessed via the Internet."
```
[the remark this answers is not in the document]
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:95]: I've not set access_config anywhere, and I've not had issues accessing yum. I ran another test to verify and was able to install with yum (slowly, but that seems to be the norm lately). [here](https://github.com/SchedMD/slurm-gcp/blob/master/terraform/slurm_cluster/modules/_slurm_instance/main.tf#L75) is where it's handled by slurm, and they do nothing to set defaults, so if it's not set by our default its just not set.

File: tools/validate_configs/test_configs/hpc-cluster-small-slurm-v5.yaml [hunk 4]
```diff
@@ -71,3 +71,11 @@ deployment_groups:
...
     - compute_partition
     - homefs
+
+  - source: community/modules/scheduler/schedmd-slurm-gcp-v5-login
+    kind: terraform
+    id: slurm_login
+    use:
+    - network1
+    - slurm_controller
+    - homefs
```
[the remark this answers is not in the document]
Comment (heyealex) [reply tools/validate_configs/test_configs/hpc-cluster-small-slurm-v5.yaml:81]: It's handled by the controller, I removed this use value. I'm also adding a task (or +1'ing an existing task) to warn if "use" modules are not actually used.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+variable "disk_type" {
+  type        = string
+  description = "Boot disk type, can be either pd-ssd, local-ssd, or pd-standard."
+  default     = "pd-standard"
+}
+
+variable "disk_size_gb" {
+  type        = number
+  description = "Boot disk size in GB."
+  default     = 100
```
[the remark this answers is not in the document]
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:263]: The image size for hpc-centos-7 is 30gb, so it has to be bigger. I'll set it to 50g to provide some space for larger images or heavy startup-scripts.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+
+variable "source_image" {
+  type        = string
+  description = "Source disk image."
+  default     = null
+}
+
+variable "disk_type" {
+  type        = string
+  description = "Boot disk type, can be either pd-ssd, local-ssd, or pd-standard."
```
[the remark this answers is not in the document]
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:256]: Validation added

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+  default     = null
+}
+
+variable "source_image_family" {
+  type        = string
+  description = "Source image family."
+  default     = null
+}
+
+variable "source_image" {
```
[the remark this answers is not in the document]
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:248]: I added a link to the slurm-gcp docs on images and a bit more clarity to the description overall.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+
+variable "can_ip_forward" {
+  type        = bool
+  description = "Enable IP forwarding, for NAT instances for example."
+  default     = false
+}
+
+variable "network_self_link" {
+  type        = string
+  description = "Network to deploy to. Only one of network or subnetwork should be specified."
```
[the remark this answers is not in the document]
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf]: I reset them to null by default and clarified the wording.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+  description = <<-EOD
+    Zone where the instances should be created. If not specified, instances will be
+    spread across available zones in the region.
+    EOD
+  default     = null
+}
+
+variable "metadata" {
+  type        = map(string)
+  description = "Metadata, provided as a map"
```
[the remark this answers is not in the document]
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf]: Fixed this and the other v5 modules

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf (hunk 1 above)
Comment (heyealex) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:1]: I have added links in a few places to clarify defaults.

Preferably defaults are listed and therefore self-documenting, but in many cases the underlying modules are setting them automatically instead. There isn't really a single solution I can see to this, as different variables behave differently.

Documenting them as we find them, with links as needed, is probably the best approach for now.

File: community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (nick-stroud) [reply community/modules/scheduler/schedmd-slurm-gcp-v5-login/variables.tf:98]: 🤷‍♂️ if it works, it works. I would expect this to cause problems but you would be seeing them under the current conditions. My experience has been with CFT modules, so maybe they are doing something else I am not aware of.

**Review by nick-stroud — APPROVED** [review]

**Pull request merged**